packages/flutter/lib/logging.dart

```diff
@@ -0,0 +1,160 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// The Flutter logging framework.
+///
+/// Logging in Flutter revolves around a familiar pattern of named logger objects
+/// that correspond to specific systems or functional areas that, using a
+/// traditional dot-separated namespace, can be further treated as hierarchical
+/// by clients (tools such as IDEs).
+///
+/// From a client's perspective, named loggers correspond to "streams" of events
+/// that tools can opt into by subscription via a VM service call.  Loggers with
+/// no subscriptions do not produce any events. Common programming idioms can
+/// make logging calls very low cost.
+///
+/// The structure of logger calls is designed to be familiar and compatible with
+/// that of `package:logging`.  Behind the scenes, logged events are forwarded
+/// to the logging facility of `dart.developer`.
+///
+/// A logger is created via constructor call:
+///
+/// ```dart
+/// final Logger _xLogger = new Logger('functional.area.x');
+/// ```
+///
+/// Messages are logged with method calls that correspond to
+/// `package:logging` levels.  For example,
+///
+/// ```dart
+/// _xLogger.info('X is starting...');
+/// ```
+///
+/// produces a log event of [Level.INFO].
+///
+/// If no client is subscribed to its corresponding stream, logging calls are
+/// ignored (and no event passed on to the `dart.developer` log function).  The
+/// cost of logging calls can be further mitigated at call sites by invoking
+/// them in a function that is only evaluated in profile or debug modes. For
+/// example,
+///
+/// ```dart
+/// profile(() {
+///   _xLogger.info('X is starting...');
+///   return true;
+/// }());
```

Comment:
the `profile` method doesn't require that you execute the closure (you can omit the trailing `()`)

Reply:
👍